src/peft/peft_model.py

```diff
@@ -101,21 +101,22 @@ class PeftModel(PushToHubMixin, torch.nn.Module):
 
     def __init__(self, model: PreTrainedModel, peft_config: PeftConfig, adapter_name: str = "default"):
         super().__init__()
-        self.base_model = model
-        self.config = getattr(self.base_model, "config", {"model_type": "custom"})
         self.modules_to_save = None
-        self.peft_config = {}
         self.active_adapter = adapter_name
         self.peft_type = peft_config.peft_type
-        if not peft_config.is_prompt_learning:
-            self.peft_config[adapter_name] = peft_config
-            self.base_model = PEFT_TYPE_TO_MODEL_MAPPING[peft_config.peft_type](
-                self.base_model, self.peft_config, adapter_name
-            )
-            self.set_additional_trainable_modules(peft_config, adapter_name)
-        else:
+
+        self._is_prompt_learning = peft_config.is_prompt_learning
+        if self._is_prompt_learning:
+            self._peft_config = {adapter_name: peft_config}
+            self.base_model = model
             self.add_adapter(adapter_name, peft_config)
+        else:
+            self._peft_config = None
```

Comment:
```suggestion
```
This is not needed anymore right?

Reply:
Yes, technically, this line could be deleted. However, I prefer to have it in situations like this because it means that the type of `self._peft_config` is `PeftConfig | None`. This can be helpful for code completion or when/if we ever enable a type checker like mypy. If we don't have this line, the type checker may not correctly recognize if we forget to check for `None`. LMK if that makes sense.